kasa/smart/modules/contact.py

```diff
@@ -0,0 +1,41 @@
+"""Implementation of contact sensor module."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ...feature import Feature
+from ..smartmodule import SmartModule
+
+if TYPE_CHECKING:
+    from ..smartdevice import SmartDevice
+
+
+class ContactSensor(SmartModule):
+    """Implementation of contact sensor module."""
+
+    REQUIRED_COMPONENT = None  # we depend on availability of key
+    REQUIRED_KEY_ON_PARENT = "open"
+
+    def __init__(self, device: SmartDevice, module: str):
+        super().__init__(device, module)
+        self._add_feature(
+            Feature(
+                device,
+                id="is_open",
+                name="Is Open",  # TODO: this hack is required until #904 gets merged
+                container=self,
+                attribute_getter="is_open",
+                icon="mdi:door",
```

Comment:
Should this be a `Primary` category and `BinarySensor`?

Reply:
Yeah, it really should :+1: